pupepat/test/pupil-synth/src/pupil_plate_image.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+from astropy.io import fits
+from src.source import Source
+
+
+class PupilPlateImage:
+    def __init__(self, shape=(4096, 4096), seed=42):
+        self.shape = shape
+        self._random = np.random.RandomState(seed)  # mine
+
+        self.sources = []
+        self._cutout_margin_factor = 1.2
+
+        # Default Noise parameters
+        self.sky_level = 200.
+        self.read_noise = self.sky_level / 100.
+        self.add_poisson_noise = True
+
+    def render(self) -> np.array:
+        """
+        Given the properties that have been set (background, sources, noise, etc)
+        generate the pupil plate image as an np.array
+
+        :return: image
+        """
+        image = np.zeros(self.shape, dtype=np.float_)
+
+        # # add the background sky level
+        image += self.sky_level
+
+        # add the sources
+        for source in self.sources:
+            # add (+=) source decal into the appropriate slice of the data
+            a = source.x - source.half_width
+            b = source.y - source.half_width
+
+            decal = source.decal
+            shape = decal.shape
+            image[a:a + shape[0], b:b + shape[1]] += decal
```

Comment:
Personal style thing: I think defining shape as a new variable makes this code more confusing. I would use 
```suggestion
            image[a:a + decal.shape[0], b:b + decal.shape[1]] += decal
```

Reply:
yes